### PR TITLE
Update dependencies, TypeHint &  Fix headers 

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Use Guzzle 7, PHPUnit 8, Symfony 4.4, and latest version of Php dev tools
| Type?             | improvement

### CODE REVIEW

**Review by Progi1984 — APPROVED** [review]

File: src/AdvancedCircuitBreaker.php [hunk 1]
```diff
@@ -119,7 +119,7 @@ public function getDefaultFallback()
      *
      * @return AdvancedCircuitBreaker
      */
-    public function setDefaultFallback(callable $defaultFallback = null)
```
Comment (atomiix) [inline src/AdvancedCircuitBreaker.php:122]: Maybe delete the phpdoc here as well?

File: src/Exception/UnsupportedMethodException.php [hunk 2]
```diff
@@ -8,11 +34,9 @@
 class UnsupportedMethodException extends CircuitBreakerException
 {
     /**
-     * @param string $methodName
-     *
      * @return UnsupportedMethodException
      */
```
Comment (atomiix) [inline src/Exception/UnsupportedMethodException.php:38]:
```suggestion
```

File: src/Storage/SymfonyCache.php [hunk 3]
```diff
@@ -71,7 +97,7 @@ public function clear()
      *
      * @return string the transaction unique identifier
      */
-    private function getKey($service)
+    private function getKey($service): string
```
Comment (atomiix) [inline src/Storage/SymfonyCache.php:100]:
```suggestion
    private function getKey(string $service): string
```

File: tests/FactorySettingsTest.php [hunk 4]
```diff
@@ -16,12 +17,11 @@
...
- * needs please refer to http://www.prestashop.com for more information.
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
  *
- * @author    PrestaShop SA <contact@prestashop.com>
- * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
- * International Registered Trademark & Property of PrestaShop SA
  */
```
Comment (atomiix) [inline tests/FactorySettingsTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/Place/HalfOpenPlaceTest.php [hunk 5]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/Place/HalfOpenPlaceTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/Place/OpenPlaceTest.php [hunk 6]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/Place/OpenPlaceTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/SimpleCircuitBreakerFactoryTest.php [hunk 7]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/SimpleCircuitBreakerFactoryTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/Storage/DoctrineCacheTest.php [hunk 8]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/Storage/DoctrineCacheTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/Storage/SimpleArrayTest.php [hunk 9]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/Storage/SimpleArrayTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/Storage/SymfonyCacheTest.php [hunk 10]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/Storage/SymfonyCacheTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/SymfonyCircuitBreakerEventsTest.php [hunk 11]
```diff
@@ -1,8 +1,32 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/SymfonyCircuitBreakerEventsTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/System/MainSystemTest.php [hunk 12]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/System/MainSystemTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/Transaction/SimpleTransactionTest.php [hunk 13]
```diff
@@ -1,16 +1,40 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/Transaction/SimpleTransactionTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/Exception/InvalidTransactionTest.php [hunk 14]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/Exception/InvalidTransactionTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/Exception/InvalidPlaceTest.php [hunk 15]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/Exception/InvalidPlaceTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/Event/TransitionEventTest.php [hunk 16]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/Event/TransitionEventTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/Client/GuzzleClientTest.php [hunk 17]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/Client/GuzzleClientTest.php:26]:
```suggestion
declare(strict_types=1);

```

File: tests/CircuitBreakerWorkflowTest.php [hunk 18]
```diff
@@ -1,4 +1,28 @@
...
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
```
Comment (atomiix) [inline tests/CircuitBreakerWorkflowTest.php:26]:
```suggestion
declare(strict_types=1);

```

**Review by Progi1984 — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Pull request merged**